modules/appstream.yaml

```diff
@@ -0,0 +1,34 @@
+name: appstream
```

Comment:
why do you need to build appstream?

Reply:
Because `appstreamcli` is not yet included in the runtime, and we need it to compose the final metadata. It is very much a build-time-only dependency :-)

Reply:
this should be done for a `make dist` or equivalent.

Reply:
or done when you write the release note and check them in prior to the release, if you don't make tarballs.

Reply:
No, it is using `appstreamcli news-to-metainfo`, which generates the final metadata from a NEWS.md file and the metainfo file. It's quite a common practice, actually, and allows you to only write your changelog once, as well as not writing it in XML which can become quite hard to read after a while. Some `make dist` like thing would only work if we would publish tarballs, not when building from a Git tag.

Reply:
"quite common" doesn't make it right. This is NOT a build time thing. This is a release time thing.

Why not using tarballs?

Reply:
Simple, because it's extra effort to implement a release procedure. And duplicating the same data in two places is bad too. Allowing this kind of implementation was actually one of the number-one feature requests for AppStream for a while, and is even documented there (but I explained it much better in a [blogpost](https://blog.tenstral.net/2020/03/maintain-release-info-easily-in-metainfo-files.html) once, the documentation really needs an overhaul).

For a project like this, the more stuff that is automated, the better. In natural science, it is really hard to keep a project maintained, because people finish their PhDs and take their knowledge with them, the project looses funding, the project was specific to only one department, etc. So, the less things people need to know to maintain a thing, the better (and creating a release tarball in addition to Git-tagging it is an extra step).
Also, it allows me to just use a commit-hash here until the final release with some more Flatpak support code is tagged and this build can use a Git tag instead of the hash :-)